src/Agent.Worker/Container/DockerCommandManager.cs

```diff
@@ -298,16 +298,19 @@ private async Task<int> ExecuteDockerCommandAsync(IExecutionContext context, str
                 }
             }
 
-            return await processInvoker.ExecuteAsync(
-                workingDirectory: HostContext.GetDirectory(WellKnownDirectory.Work),
-                fileName: DockerPath,
-                arguments: arg,
-                environment: null,
-                requireExitCodeZero: false,
-                outputEncoding: null,
-                killProcessOnCancel: false,
-                redirectStandardIn: redirectStandardIn,
-                cancellationToken: cancellationToken);
+            using (redirectStandardIn)
```

Comment:
this using does not acquire any resources. 
In this case I would say directly calling .Dispose() is a better pattern.

Reply:
what happens if you pass this dispable to the await process?

Reply:
I did the using() since we were returning and thought it was easier to read than assigning to a temp variable, calling Dispose() then returning, but I am fine changing it if that looks better to others